docs/reference/pip.rst

```diff
@@ -90,6 +90,25 @@ before invoking ``setup.py``. The injection should be transparent to
 ``setup.py`` emulating the commands pip requires may need to be aware that it
 takes place.
 
+Build System Output
+~~~~~~~~~~~~~~~~~~~
+
+Any output produced by the build system will be read by pip (for display to the
+user if requested). In order to correctly read the build system output, pip
+requires that the output is written in a well-defined encoding, specifically
+the encoding returned by python's ``locale.getpreferredencoding`` function, or
+"utf8" if ``getpreferredencoding`` does not return a value (or returns "ASCII",
+which ).
```

Comment:
The most reliable incantation I know for detecting ASCII as the preferred encoding is: `codecs.lookup(locale.getpreferredencoding()).name == "ascii"`

The reason for that is that going through the codec system and then looking at the *codec* name automatically deals with the fact that different platforms may report the same encoding under different aliases. Most importantly for this purpose, some Linux systems report ASCII by its official spec number, 'ANSI_X3.4-1968'.

(The patch itself handles this correctly, but it makes sense to include it here for the benefit of build tool developers as well)